packages/miniflare/src/workers/kv/assets.worker.ts

```diff
@@ -13,21 +17,26 @@ export default <ExportedHandler<Env>>{
 			return new Response(message, { status: 405, statusText: message });
 		}
 
-		// Decode key
-		const url = new URL(request.url);
-		let key = url.pathname.substring(1); // Strip leading "/"
+		const decoder = new TextDecoder();
+		const reverseMap: AssetReverseMap = JSON.parse(
+			decoder.decode(env.__STATIC_ASSETS_REVERSE_MAP)
+		);
 
-		if (url.searchParams.get(KVParams.URL_ENCODED)?.toLowerCase() === "true") {
-			key = decodeURIComponent(key);
-		}
+		// don't uri decode pathname, because we encode the filepath before hashing
+		const key = new URL(request.url).pathname.substring(1);
+
+		const entry = reverseMap[key] ?? {};
+		const filePath = entry["filePath"] ?? "";
+		// falls back to application/octet-stream
+		// const contentType = entry["contentType"];
 
 		const blobsService = env[SharedBindings.MAYBE_SERVICE_BLOBS];
-		if (key === "" || key === "/") {
+		if (filePath === "" || filePath === "/") {
 			return new Response("Not Found", {
 				status: 404,
 			});
```

Comment:
Actually how about:

```ts
const entry = reverseMap[key];
if ( entry === undefined ) {
  return new Response("No Found", { status: 404 });
}
const { filePath, contentType } = entry;
assert(filePath, "Invalid `filePath` in `reverseMap`");
```

Reply:
i don't think it even needs to check for filepath right, since it shouldn't be possible for an entry to exist without a filepath/this should error earlier when constructing the reverse map